### PR TITLE
http2: add test for frames

### PR DESCRIPTION
**Pull request by catenacyber**
## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/5743

### CODE REVIEW

File: tests/http2-frames/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  min-version: 8
+
+# disables checksum verification
+args:
+  - -k none --set stream.midstream=true
+
+pcap:  ../http2-keywords2/input.pcap
+
+checks:
+# checks for http.uri keyword : 1 for HTTP1, 1 for mimicked HTTP2 response, so 2 for whole HTTP
```
Comment (inashivb) [inline tests/http2-frames/test.yaml:11]: leftover from copy paste?

File: tests/http2-frames/test.rules [hunk 2]
```diff
@@ -0,0 +1 @@
+alert http2 any any -> any any (frame:http2.hdr; content:"|00 00 04 08 00|"; flow:to_server; sid:1;)
```
Comment (inashivb) [inline tests/http2-frames/test.rules:1]: Why don't we test all frame types?

File: tests/eve-payload-07-http-gap/suricata.yaml [hunk 3]
```diff
@@ -7,6 +7,9 @@ outputs:
       filetype: regular #regular|syslog|unix_dgram|unix_stream|redis
       filename: eve.json
       types:
+        - frame:
+            # this should not affect detection (but it did)
```
Comment (inashivb) [inline tests/eve-payload-07-http-gap/suricata.yaml:11]: This really shouldn't. This is output config section..
Q: Is the fix really then https://github.com/OISF/suricata/pull/11717/commits/ec69ad0a7a6d1918839b00c75314af68aa013765 ?
It seems to me that stream frame should be enabled at all times. wdyt?

File: tests/http2-frames/test.yaml (hunk 1 above)
Comment (catenacyber) [reply tests/http2-frames/test.yaml:11]: indeed removing

File: tests/eve-payload-07-http-gap/suricata.yaml (hunk 3 above)
Comment (catenacyber) [reply tests/eve-payload-07-http-gap/suricata.yaml:11]: > This really shouldn't. 

Indeed, it should not affect detection but it does now in master...
The fix is indeed https://github.com/OISF/suricata/commit/ec69ad0a7a6d1918839b00c75314af68aa013765

> It seems to me that stream frame should be enabled at all times. wdyt?

Suricata is smart enough that frames are truly created only if they get used in rules or output...
So, no rules with frame:http2.stream, and no output, means these frames do not get created...

File: tests/http2-frames/test.rules (hunk 2 above)
Comment (catenacyber) [reply tests/http2-frames/test.rules:1]: :-p

Because I tested the one I needed

Adding the other ones

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #2036, thanks!

**Pull request closed without merging**